packages/wordcount/src/defaultSettings.js

```diff
@@ -1,3 +1,32 @@
+/**
+ * @typedef {import("./index").WordCountStrategy} WordCountStrategy
+ */
+
+/**
+ * @typedef {Object} WordCountSettingsFields
```

Comment:
What I've generally observed is that the "Object" type is interpreted by TypeScript in a way which isn't expected, and often results in types not being enforced (i.e. inferred as `any`).

>Don’t ever use the types Number, String, Boolean, Symbol, or Object These types refer to non-primitive boxed objects that are almost never used appropriately in JavaScript code.

See also: https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#general-types

Through the course of discussion in #20427, it was found that `Record` tends to work pretty well where we might otherwise reach to `Object`.

Documentation: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/coding-guidelines.md#generic-types

In this case, and as noted elsewhere in the documentation, you can omit `{Object}` altogether for `@typedef` that you intend to ascribe properties:

>Note that there is no {Object} between @typedef and the type name. As @propertys below tells us that it is a type for objects, it is recommend to not use {Object} when you want to define types for your objects.

https://github.com/WordPress/gutenberg/blob/master/docs/contributors/coding-guidelines.md#custom-types

```suggestion
 * @typedef WordCountSettingsFields
```

Reply:
Good catch, @aduth. I think this was remnant of another iteration and now it’s super obvious I glossed over the “Custom Types” section of the guidelines :) Fixed in 51e005b.